### PR TITLE
numcodecs compression (#60)

### PR DESCRIPTION
**Pull request by marklescroart**
* Added numcodecs compression options
* Backward compatibility for download of raw arrays w/ gzip=True
* Modify download_raw_array to be memory-efficient while decompressing numcodecs arrays

All changes from prev PR from forked repo, now here. All seems to work well, no memory overhead.

### CODE REVIEW

**Comment by anwarnunez** [comment]
Very cool, thanks!
It looks like there is on missing `numcodec` import still:
https://travis-ci.org/gallantlab/cottoncandy/jobs/403460114

File: cottoncandy/interfaces.py [hunk 1]
```diff
@@ -599,10 +617,17 @@ def upload_raw_array(self, object_name, array, gzip=True, acl=DEFAULT_ACL, **met
             zipdata.seek(0)
             filestream = zipdata
             data_nbytes = get_fileobject_size(filestream)
-        else:
+        elif hasattr(numcodecs, compression.lower()): # if the mentioned compression type is in numcodecs
```
Comment (anwarnunez) [inline cottoncandy/interfaces.py]: Seems this logic can be more transparently handled by using:
`numcodecs.get_codec(dict(id=compression.lower())`
I can work on this later this week.

**Comment by marklescroart** [comment]
Wasn't missing import, was missing numcodecs in requirements.txt. This push should fix it...

File: cottoncandy/interfaces.py (hunk 1 above)
Comment (marklescroart) [reply cottoncandy/interfaces.py]: Addressed in latest push.

File: cottoncandy/interfaces.py [hunk 2]
```diff
@@ -42,6 +42,11 @@
                     GzipInputStream,
                     DEFAULT_ACL, MPU_CHUNKSIZE, MPU_THRESHOLD, DASK_CHUNKSIZE, MB, SEPARATOR,
                     MAGIC_CHECK)
+from .options import config
+
+DO_COMPRESSION = config.get('compression', 'do_compression').lower() in ('true', 't', 'y','yes')
```
Comment (anwarnunez) [inline cottoncandy/interfaces.py]: you can use `cottoncandy.utils.string2bool` to convert strings to booleans:
https://github.com/gallantlab/cottoncandy/blob/master/cottoncandy/utils.py#L67

**Pull request merged**